pkg/frontend/transport/handler.go

```diff
@@ -192,7 +192,7 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			} else {
 				err = httpgrpc.Errorf(http.StatusBadRequest, err.Error())
```

Comment:
We don't need this anymore if using `http.Error`. Same as `errRequestEntityTooLarge` I don't think we need to wrap the error with grpc.
We can just return the raw error. But you need to check the status code.

```
statusCode := http.StatusBadRequest
if err := r.ParseForm(); err != nil {
  if util.IsRequestBodyTooLarge(err) {
    statusCode = http.StatusRequestEntityTooLarge
  }
  http.Error(w, err.Error(), statusCode)
}
```